cmd/tg_bot/internal/handlers/text.go

```diff
@@ -0,0 +1,181 @@
+package handlers
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/pkg/errors"
+	tele "gopkg.in/telebot.v3"
+	"nodemon/cmd/tg_bot/internal"
+	"nodemon/pkg/messaging/pair"
+)
+
+func AddNewNodeHandler(
+	c tele.Context,
+	environment *internal.TelegramBotEnvironment,
+	requestType chan pair.RequestPair,
+	responsePairType chan pair.ResponsePair) error {
+
+	url := strings.TrimPrefix(c.Text(), "Add ")
+	if !strings.HasPrefix(url, "http") && !strings.HasPrefix(url, "https") {
+		return c.Send(
+			"Sorry, the url seems to be incorrect",
+			&tele.SendOptions{ParseMode: tele.ModeDefault},
+		)
+	}
+	requestType <- &pair.InsertNewNodeRequest{Url: url}
+
+	response := fmt.Sprintf("New node was '%s' added", url)
+	err := c.Send(
+		response,
+		&tele.SendOptions{ParseMode: tele.ModeHTML})
+	if err != nil {
+		return nil
+	}
+	urls, err := requestNodesList(requestType, responsePairType)
+	if err != nil {
+		return errors.Wrap(err, "failed to request nodes list buttons")
+	}
+	message, err := environment.NodesListMessage(urls)
+	if err != nil {
+		return errors.Wrap(err, "failed to construct nodes list message")
+	}
+	return c.Send(
+		message,
+		&tele.SendOptions{
+			ParseMode: tele.ModeHTML,
+		},
+	)
+}
+
+func RemoveNodeHandler(
+	c tele.Context,
+	environment *internal.TelegramBotEnvironment,
+	requestType chan pair.RequestPair,
+	responsePairType chan pair.ResponsePair) error {
+
+	url := strings.TrimPrefix(c.Text(), "Remove ")
+	if !strings.HasPrefix(url, "http") && !strings.HasPrefix(url, "https") {
+		return c.Send(
+			"Sorry, the url seems to be incorrect",
+			&tele.SendOptions{ParseMode: tele.ModeDefault},
+		)
+	}
```

Comment:
Use `url.Parse` for URL checking.

Reply:
I don't really see how it could be useful. The function returns an object where the `host` is separated from the `scheme` and what is more, it does not really gurantee that the host was either `http` or `https`. I tried to put wrong name like `ht://mainet` it the function did not return an error. So even using this function I would have to write smth like this:
```go
	node := "http://mainnet-statehash-aws-ir-4.wavesnodes.com"
	u, err := url.Parse(node)
	if err != nil {
		return err
	}
	if u.Host != "http" && u.Host != "https" {
		return
	}
```
It just increases the amount of code for simple check of having "http" or "https"

Reply:
I think [this](https://stackoverflow.com/questions/31480710/validate-url-with-standard-package-in-go) will be helpful. And you're right, `url.Parse` isn't right instrument for URL checking.

Reply:
It is not helpful as well. It does not fail with a prefix like "htt". Anyway, I am going to get rid of `http` and `https` prefixes in the next PR, so let's skip it for now

Reply:
Okay